log-ballerina/natives.bal

```diff
@@ -103,82 +116,65 @@ public isolated function printInfo(string msg, *KeyValues keyValues, error? 'err
 # + keyValues - The key-value pairs to be logged
 # + 'error - The error struct to be logged
 public isolated function printWarn(string msg, *KeyValues keyValues, error? 'error = ()) {
-    if (isLogLevelEnabledExtern(WARN)) {
+    if (isLogLevelEnabled(WARN)) {
         print(WARN, msg, keyValues, 'error);
     }
 }
 
 isolated function print(string logLevel, string msg, *KeyValues keyValues, error? err = ()) {
-    string keyValuesString = "";
+    LogRecord logRecord = {
+        time: getCurrentTime(),
+        level: logLevel,
+        module: getModuleName() == "." ? "" : getModuleName(),
+        message: msg
+    };
+    if err is error {
+        logRecord["error"] = err.message();
+    }
     foreach [string, Value] [k, v] in keyValues.entries() {
         anydata value;
         if (v is Valuer) {
             value = v();
         } else {
             value = v;
         }
-        keyValuesString += appendKeyValue(k, value);
+        logRecord[k] = value;
     }
-    printExtern(logLevel, getOutput(msg, keyValuesString, err), format);
-}
-
-isolated function printExtern(string logLevel, string msg, string outputFormat) = @java:Method {
-    'class: "org.ballerinalang.stdlib.log.Utils"
-} external;
-
-isolated function appendKeyValue(string key, anydata value) returns string {
-    string k;
-    string v;
-    if (format == JSON_OUTPUT_FORMAT) {
-        k = string `, "${key}": `;
+    if format == "json" {
+        printJsonExtern(logRecord.toJsonString());
     } else {
-        k = string ` ${key} = `;
+        printLogFmtExtern(logRecord);
     }
-    if (value is string) {
-        v = string `"${value}"`;
-    } else {
-        v = value.toString();
-    }
-    return k + v;
 }
 
-isolated function getOutput(string msg, string keyValues, error? err = ()) returns string {
-    string output = "";
-    if (format == JSON_OUTPUT_FORMAT) {
-        output = string `"message": ${getMessage(msg, err)}${keyValues}`;
-    } else {
-        output = string `message = ${getMessage(msg, err)}${keyValues}`;
-    }
-    return output;
+public isolated function printJsonExtern(string msg) {
+    println(err(), java:fromString(msg));
 }
```

Comment:
Do we need to make this public? and also can we directly call `println` function like,
```
if format == "json" {
    println(err(), java:fromString(logRecord.toJsonString()));
}
```

Reply:
Fixed in 401ad55